docs/sql-migration-guide-upgrade.md

```diff
@@ -50,6 +50,8 @@ license: |
 
   - In Spark version 2.4 and earlier, JSON datasource and JSON functions like `from_json` convert a bad JSON record to a row with all `null`s in the PERMISSIVE mode when specified schema is `StructType`. Since Spark 3.0, the returned row can contain non-`null` fields if some of JSON column values were parsed and converted to desired types successfully.
 
+  - Refreshing a cached table would trigger a table uncache operation and then a table cache (lazily) operation. In Spark version 2.4 and earlier, the cache name ane cache storage level are not preserved before the uncache operation. Therefore, the cache name and cache storage level could be changed unexpectedly. Since Spark 3.0, cache name and cache storage level will be first preserved for cache recreation. It helps to maintain a consistent cache behavior upon table refreshing.
```

Comment:
does user need to know it when upgrading Spark?

Reply:
It's a good question -- I had assumed the cached name is visible to the user, but is that true? if it is it's worth a release note

Reply:
I think it's visible via the web UI, but every bug fix is visible to users, I think migration guide is for the behavior changes that might surprise users?

Reply:
I agree, and if we make too many release notes, the important ones might get lost. If it's merely cosmetic, but is user-visible, I'm on the fence about it. @gatorsmile is this a case where you'd prefer a migration note?

Reply:
Hi @gatorsmile, please kindly let us know if you prefer a migration note to this PR. If not, I may remove the change accordingly. Many thanks.

Reply:
I would like to keep this in the upgrade guide. I think we should have it.

Reply:
Hi @gatorsmile , thanks for confirming this. Btw, would you also mind to review this PR again?